src/codegen/compile.ml

```diff
@@ -1622,20 +1622,22 @@ module Tagged = struct
   let alloc env size tag =
     assert (size > 1l);
     let name = Printf.sprintf "alloc_size<%d>_tag<%d>" (Int32.to_int size) (Int32.to_int (int_of_tag tag)) in
-    (* Computes a (conservative) mask for the bumped HP, so that the existence of non-zero bits under it
+    (* Computes a mask for the bumped HP, so that the existence of non-zero bits under it
```

Comment:
If I am not mistaken, the mask is still conservative, isn't it? E.g. if we allocate 12, then 18 on a new partition, it is taking two times the slow path?

Reply:
I don't think so. E.g. allocating 28 first will be detected due to its own mask (it is not a power of 2):
```
(lldb) p/x (0x3333ffef + 28) & 0xffe0
(int) $6 = 0x00000000
```

Reply:
I only called the mask conservative because (as you noted in the other PR) it didn't detect that increments `0xPPPP0000` -> `0xPPPP0800` (powers of 2) remained on-page. The situation is now that with spewed HP and extended pow-of-2 masks both _page-hops_ and _remains_ get detected accurately.

Reply:
Yes, indeed. I only wondered whether there are still cases where a sequence of allocations on the same page take the slow path, e.g. 12, then 48. It is not about correctness - these cases still check the page bounds when needed. I am curious about whether the mask extension is general in the sense that we only perform the slow path when necessary.